File: example_code/mpi/MPI_Code/Fortran/scatterv.f90
PROGRAM scatterv
USE mpi_f08
IMPLICIT NONE
INTEGER::isend(6), irecv(3)=0, isndcnt(0:2), idisp(0:2)
INTEGER::rcvcnt,myrank
isend=(/1,2,2,3,3,3/)
isndcnt=(/1,2,3/)
idisp=(/0,1,3/)
CALL MPI_Init
CALL MPI_Comm_rank(MPI_COMM_WORLD, myrank)
rcvcnt=myrank+1
CALL MPI_Scatterv(isend,isndcnt,idisp,MPI_INTEGER,irecv,rcvcnt,MPI_INTEGER,0,MPI_COMM_WORLD)
PRINT*,'rank: ',myrank, 'irecv=',irecv

irecv=0
isend=(/10,20,20,30,30,30/)
IF(myrank==0)THEN
  irecv(1)=isend(1)
  CALL MPI_Scatterv(isend,isndcnt,idisp,MPI_INTEGER,MPI_IN_PLACE,rcvcnt,MPI_INTEGER,0,MPI_COMM_WORLD)
ELSE
  CALL MPI_Scatterv(isend,isndcnt,idisp,MPI_INTEGER,irecv,rcvcnt,MPI_INTEGER,0,MPI_COMM_WORLD)
ENDIF
PRINT*,'RANK: ',myrank, 'IRECV=',irecv
CALL MPI_Finalize
END PROGRAM scatterv

File: example_code/mpi/MPI_Code/C/scatterv.c
#include <stdio.h>
#include <mpi.h>
int main()
{
  int i, myrank;
  int isend[6]={1,2,2,3,3,3}, irecv[3]={0,};
  int isndcnt[3]={1,2,3}, idisp[3]={0,1,3};
  int rcvcnt;
  MPI_Init(NULL,NULL);
  MPI_Comm_rank(MPI_COMM_WORLD,&myrank);
  rcvcnt=myrank+1;
  MPI_Scatterv(isend,isndcnt,idisp, MPI_INT, irecv, rcvcnt,MPI_INT, 0, MPI_COMM_WORLD);
  printf("rank : %d    irecv=",myrank);
  for(i=0;i<3;i++) printf("%d ",irecv[i]);
  printf("\n");

  for(i=0;i<3;i++) irecv[i]=0;
  if(myrank==0){
    isend[0]=10, isend[1]=20, isend[2]=20;
    isend[3]=30, isend[4]=30, isend[5]=30;
    irecv[0]=isend[0];
    MPI_Scatterv(isend,isndcnt,idisp,MPI_INT,MPI_IN_PLACE,rcvcnt,MPI_INT,0,MPI_COMM_WORLD);
  }
  else{
    MPI_Scatterv(isend,isndcnt,idisp, MPI_INT, irecv, rcvcnt,MPI_INT, 0, MPI_COMM_WORLD);
  }
  printf("RANK : %d    IRECV=",myrank);
  for(i=0;i<3;i++) printf("%d ",irecv[i]);
  printf("\n");

  MPI_Finalize();
  return 0;
}
